### ISSUE TITLE
Carousel: minor a11y fixes

### ISSUE THREAD

**Issue by ianmcburnie** [eBay/ebayui-core#294]
# Bug Report

## Description

A few updates needed to the markup for accessibility (a11y).

### 1. Fix Live Region for Voiceover (fixed in #297)

An additional span is required inside live region:

```html
<span id="carousel-status-c1-1" class="clipped" role="status" aria-live="polite">Showing slide 1 of 4 - Top Products - Carousel</span>
```

should be:

```html
<span id="carousel-status-c1-1" class="clipped" role="status" aria-live="polite">
  <span>Showing slide 1 of 4 - Top Products - Carousel</span>
</span>
```

Reason: this is required to trigger a11y subtree change & announcement in voiceover.

### 2. Add Support for Clipped Heading

We should also allow the option of this outer span being a heading element too (h1 - h6), in the case where the page/app does not have a visible heading preceding the carousel.

```html
<span id="carousel-status-c1-1" class="clipped" role="status" aria-live="polite">
  <span>Showing slide 1 of 4 - Top Products - Carousel</span>
</span>
```

Could be:

```html
<h2 id="carousel-status-c1-1" class="clipped" role="status"  aria-live="polite">
  <span>Showing slide 1 of 4 - Top Products - Carousel</span>
</h2>
```

Reason: all carousels should be preceded by a heading.

### 3. Remove Carousel Pagination Landmark (fixed in #297)

Remove role="navigation".

```html
<ul class="carousel__dots" role="navigation">
```

Should be:

```html
<ul class="carousel__dots">
```

Reason: we do not wish to pollute the screenreader landmark list with the carousel dots (they do not feel important enough to be a landmark)

## 4. Add Carousel Pagination Current Item

Add `aria-current="true"` to the *current* dot.

Reason: conveys to AT that this is the current item.

Note: this change might allow us to drop the `carousel__dot--active` class (we can use an attribute selector instead)

**Comment by sendlo** [comment]
3 of 4 issues fixed.  I opened another ticket to track the final issue:
https://github.com/eBay/ebayui-core/issues/325
